src/Symfony/Component/Security/Http/Firewall.php

```diff
@@ -53,13 +54,21 @@ public function onKernelRequest(GetResponseEvent $event)
             $exceptionListener->register($this->dispatcher);
         }
 
+        $hasResponse = false;
+
         // initiate the listener chain
         foreach ($listeners as $listener) {
-            $listener->handle($event);
+            if ($listener instanceof LogoutListener) {
+                $listener->handle($event);
 
-            if ($event->hasResponse()) {
                 break;
+            } elseif ($hasResponse) {
+                continue;
             }
+
+            $listener->handle($event);
+
+            $hasResponse = $event->hasResponse();
```

Comment:
this means that it won't stop at the first listener setting the response anymore, allowing others to set the response

Reply:
Nope only `LogoutListener` can override the response if it has been set. And it is called so https://github.com/symfony/symfony/issues/7104#issuecomment-278625482 doesn't apply.

Reply:
Can you add a test case for this? with multiple listeners if possible

Reply:
I'll try but I don't know how to do it yet